### PR TITLE
Test issue 94 upload errata

### PR DESCRIPTION
**Pull request by ragabala**
This test case address the following

1) Create,sync, publish a new repo
2) Add a modular errata to it
3) Check whether the errata has been applied to the repo

Refer #94.

This PR also contains the changes that are done as a part of PR 106

### CODE REVIEW

**Comment by kersommoura** [comment]
@rochacbruno, any comments?

File: pulp_2_tests/constants.py [hunk 1]
```diff
@@ -130,6 +130,26 @@
...
+    'context': 'deadbeef',
+    'description': 'Walrus Erratum; metadata-only',
+    'from': 'betelgeuze',
+    'issued': '2002-02-20 00:22:00',
+    'rpm_name': 'walrus',
+    'stream_name': '5.21',
+    'updated': '2018-09-05 11:23:42',
+    'version': '20180704144203',
+})
+
```
Comment (kersommoura) [inline pulp_2_tests/constants.py]: Perhaps, add a description about this constant, or even rename it. With modularity, and modules being added this be misleading. We will need to make a clear distinction about what data we are mentioning. This constant is more related to the errata inside the `updateinfo.xml`.

File: pulp_2_tests/tests/rpm/api_v2/test_modular_errata.py [hunk 2]
```diff
@@ -139,13 +150,102 @@ def test_collection_field(self):
...
+        This Test does the following.
+        1. Create, sync, publish a repo with RPM_Unsigned_feed_url.
+        2. Upload a custom modular erratum to the repo. The custom
+           module erratum is obtained from ``_get_erratum()``.
+        3. Verify whether the uploaded erratum is present in the repo.
+        """
+        # Step 1
+        body = gen_repo(
+            importer_config={'feed': RPM_UNSIGNED_FEED_URL},
+            distributors=[gen_distributor(auto_publish=True)]
```
Comment (kersommoura) [inline pulp_2_tests/tests/rpm/api_v2/test_modular_errata.py]: Evaluate just publish the repo after the errata is upload as described in #94.

File: pulp_2_tests/tests/rpm/api_v2/test_modular_errata.py [hunk 3]
```diff
@@ -139,13 +150,102 @@ def test_collection_field(self):
...
+        repo_initial = self.client.get(
+            repo_initial['_href'],
+            params={'details': True}
+        )
+
+        # Step 2
+        unit = self._gen_modular_errata()
+        upload_import_erratum(self.cfg, unit, repo_initial)
+        repo = self.client.get(repo_initial['_href'], params={'details': True})
+        publish_repo(self.cfg, repo)
```
Comment (kersommoura) [inline pulp_2_tests/tests/rpm/api_v2/test_modular_errata.py]: I looked into the steps described into the issue #94. It was used `--force-full`.

```python
        publish_repo(
            self.cfg,
            repo,
            {
                "id": repo["distributors"][0]["id"],
                "override_config": {"force_full": True},
            },
        )


```

File: pulp_2_tests/tests/rpm/api_v2/test_modular_errata.py (hunk 2 above)
Comment (kersommoura) [reply pulp_2_tests/tests/rpm/api_v2/test_modular_errata.py]: @ragabala, I confirmed with @dparalen, that the repo was synced, published then the errata was uploaded.  It is fine to keep the `auto_publish` as True.

**Comment by rochacbruno** [comment]
LGTM

**Pull request merged**